### ISSUE TITLE
Issue with using Fetch URL object losing headers

### ISSUE THREAD

**Issue by kenyorkDSg** [elastic/apm-agent-rum-js#1288]
I am currently using elastic/apm-rum version 4.4.4, in the process of upgrading to 5.12.0, I see the issue with both versions.  

If I try to use the URL object as parameters to a fetch call, I am losing any custom headers that were initially sent.  If I force the URL object to a string, then the code will work fine.  It appears when the apm agent intercepts the fetch calls, it is looking for a string and not the URL object and doesn't send the headers along to the actual call.  
```
const url = new URL("http://localhost:3006/testFetch")
const response = await fetch(url, {
  headers: {
	  Authorization: `Bearer testtest`,
  },
})
```

If I were to add the toString, it will work properly

```
const url = new URL("http://localhost:3006/testFetch")
const response = await fetch(url.toString(), {
  headers: {
	  Authorization: `Bearer testtest`,
  },
})
```

Although adding toString will work, it's not ideal, this is used by a bunch of teams and that won't know to add the toString() to all their calls.  Thanks so much.

**Comment by devcorpio** [comment]
Hi @kenyorkDSg,

Thanks for using the RUM agent!

At present, the agent doesn't fully handle a fetch when the first argument is URL object. It handles the following scenarios:

- string
- [new Request(input, options)](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request)


Unfortunately and since this is something that is still pending to develop I cannot give you a workaround different to the one that you proposed previously (`url.toString()`)

I would like to ask you a few questions given that your answers could give us invaluable information to keep in mind when developing this enhancement:

- Why are you using new URL? 
- What is the use case that URL is helping you with?


Many thanks,
Alberto

**Comment by devcorpio** [comment]
Hi @kenyorkDSg,

Thank you for providing these details, they are very helpful!

Cheers,
Alberto

**Comment by devcorpio** [comment]
Hi @kenyorkDSg,

We just [released](https://www.elastic.co/guide/en/apm/agent/rum-js/current/release-notes-5.x.html) this fix. Available in 5.14.0

Thanks,
Alberto
